examples/tutorial/jupyter/execution/pandas_on_unidist/test/test_notebooks.py

```diff
@@ -0,0 +1,123 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+import os
+import sys
+
+import nbformat
+
+MODIN_DIR = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), *[".." for _ in range(6)])
+)
+sys.path.insert(0, MODIN_DIR)
+from examples.tutorial.jupyter.execution.test.utils import (  # noqa: E402
+    _replace_str,
+    _execute_notebook,
+    test_dataset_path,
+    download_taxi_dataset,
+    change_kernel,
+)
+
+# the kernel name "python3mpi" must match the one
+# that is set up in `examples/tutorial/jupyter/execution/pandas_on_unidist/setup_kernel.py`
+# for `Unidist` engine
+change_kernel(kernel_name="python3mpi")
```

Comment:
I prefer to use my suggestion in order not to create the executor twice.

Reply:
Unfortunately, it does not work, because the ExecutePreprocessor is created before the modin configuration can be changed. In our example, we are changing modin configuration after starting jupyter notebook, but the kernel should 
 already be running at that moment. If this is duplicated in the test, it will corrupt the scenario and not check if we set it in notebook.
 
 I have added an executor preprocessor cache to avoid creating a lots of preprocessors.